src/Workspaces/Remote/ServiceHub/xlf/ServiceHubResources.fr.xlf

```diff
@@ -0,0 +1,42 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="en" target-language="fr" original="../ServiceHubResources.resx">
+    <body>
+      <trans-unit id="References">
+        <source>References</source>
+        <target state="new">References</target>
+        <note />
+      </trans-unit>
+      <trans-unit id="This_0_has_1_references">
+        <source>This {0} has {1} reference(s)</source>
+        <target state="new">This {0} has {1} reference(s)</target>
```

Comment:
in general you can't really create strings in this manner.  Depending on the noun, the different languages may need an entirely different string to present to the user.

Reply:
we got it from existing one. but sure, we can change it if there is better way. can you give more detail?

Reply:
Sure.  the loc string would either have to be something lik  "'{0}' has {1} reference(s)".  This would indicate that the '0' guy is just a thing that has no other relation to any of the other words and can be substituted with any noun.  In general, this can be localized just fine.  *Or*, you need a specific loc string for each type of thing you would put in that spot.  i.e.:

"This method has {0} reference(s)" and "This type has {0} reference(s)".

The reason for thsi is that in different languages, you might need to localize each of these very differently.  For example, 'method' and 'type' might be masculine or feminine in some languages, requiring very different surrounding words or sentence structures.  By just inserting the word *as is* (without anything like quotes around it), we are effectively just building a string manually, and that isn't something you can't do safely with different sorts of replacements words in different languages.

Reply:
@lineralpha so this is original - http://index/?leftProject=Microsoft.VisualStudio.CodeSense.ReferencesProvider&file=ProviderResources.resx&line=120

should we do what @CyrusNajmabadi said? have we had any feedbacks on it? tagging @jinujoseph as well.

Reply:
No. I didn't hear any feedback on it. But that might be just because CodeLens is only available on Pro+ SKUs and most of the users are English-speaking I guess.